boa/src/object/gcobject.rs

```diff
@@ -211,111 +316,10 @@ impl GcObject {
     pub fn construct(
         &self,
         args: &[Value],
-        new_target: Value,
+        new_target: &Value,
```

Comment:
Any reason here to pass a `Value` by reference? Wasn't it a GC value that can be cheaply cloned? Maybe @HalidOdat or @RageKnify can give some insights.

Reply:
I think it's fine to use `&Value`, should occupy less space while being passed as an argument and we can still clone when necessary.

Reply:
This change also makes it consistent with `call` which used `&Value`.